frappe/core/report/database_storage_usage_by_tables/database_storage_usage_by_tables.js

```diff
@@ -3,4 +3,44 @@
 
 frappe.query_reports["Database Storage Usage By Tables"] = {
 	filters: [],
+	onload: function (report) {
+		report.page.add_inner_button(
+			__("Optimize"),
+			function () {
+				let d = new frappe.ui.Dialog({
+					title: "Optimize Doctype",
+					fields: [
+						{
+							label: "Select a DocType",
+							fieldname: "doctype_name",
+							fieldtype: "Link",
+							options: "DocType",
```

Comment:
Filter out single, virtual doctypes. They don't have tables.

Reply:
Missed it 😅